tests/Doctrine/Tests/ORM/Functional/SQLFilterWithFetchEAGERTest.php

```diff
@@ -0,0 +1,284 @@
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\Query\Filter\SQLFilter;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class SQLFilterWithFetchEAGERTest extends OrmFunctionalTestCase
+{
+    /**
+     * @var int
+     */
+    private $company1Id;
+
+    /**
+     * @var int
+     */
+    private $patient1Id;
+
+    /**
+     * @var int
+     */
+    private $insurance1Id;
+
+    /**
+     * @var int
+     */
+    private $company2Id;
+
+    /**
+     * @var int
+     */
+    private $patient2Id;
+
+    /**
+     * @var int
+     */
+    private $insurance2Id;
+
+    private function createModels(): void
+    {
+        $company      = new Company();
+        $patient      = new Patient($company);
+        $insurance    = new Insurance($company, 'name1');
+        $patInsurance = new PatientInsurance($patient, $insurance, 'test1');
+
+        $company2      = new Company();
+        $patient2      = new Patient($company2);
+        $insurance2    = new Insurance($company2, 'name2');
+        $patInsurance2 = new PatientInsurance($patient2, $insurance2, 'test1');
+
+        $this->_em->persist($company);
+        $this->_em->persist($patient);
+        $this->_em->persist($patInsurance);
+        $this->_em->persist($insurance);
+
+        $this->_em->persist($company2);
+        $this->_em->persist($patient2);
+        $this->_em->persist($patInsurance2);
+        $this->_em->persist($insurance2);
+
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $this->company1Id   = $company->id;
+        $this->patient1Id   = $patient->id;
+        $this->insurance1Id = $insurance->id;
+
+        $this->company2Id   = $company2->id;
+        $this->patient2Id   = $patient2->id;
+        $this->insurance2Id = $insurance2->id;
+    }
+
+    public function testCase(): void
+    {
+        $this->createSchemaForModels(Company::class, Patient::class, PatientInsurance::class, Insurance::class);
+        $this->_em->getConfiguration()->addFilter('company', CompanyFilter::class);
+
+        $this->createModels();
+
+        $this->_em->getFilters()->enable('company')->setParameter('companyId', $this->company1Id);
+
+        $company1 = $this->_em->find(Company::class, $this->company1Id);
+        $company2 = $this->_em->find(Company::class, $this->company2Id);
+
+        self::assertInstanceOf(Company::class, $company1);
+        self::assertNull($company2);
+
+        $patient1 = $this->_em->find(Patient::class, $this->patient1Id);
+        $patient2 = $this->_em->find(Patient::class, $this->patient2Id);
+
+        self::assertInstanceOf(Patient::class, $patient1);
+        self::assertNull($patient2);
+
+        $patInsurance = $patient1->patInsurance->toArray();
+        self::assertCount(1, $patInsurance);
+        self::assertEquals($this->insurance1Id, $patInsurance[0]->insurance->id);
+
+        $this->_em->clear();
+        $this->_em->getFilters()->enable('company')->setParameter('companyId', $this->company2Id);
```

Comment:
This looks like you're testing different scenarios here. Can we split those into separate tests please?

Reply:
it was meant to be